### ISSUE TITLE
Standalone installer\portable version

### ISSUE THREAD

**Issue by koutoftimer** [koutoftimer/sqrc#1]
Create standalone executables that will have no requirements for targeted OS.

**Comment by koutoftimer** [comment]
Ability to download the utility for `python` users from `pip` is one of the advantages of current solution, but it doesn't feel like native application for desktop users. 

Because of problems with `PyInstaller` and `py2exe` I decided to write it with `Qt` because of support of [clipboard][qt-clipboard] and ability to [render `*.svg`][qt-svgrendering].  To make `*.svg` I decided to use [nayuki/QR-Code-generator][qr-code-gen]

[qt-clipboard]: https://doc.qt.io/archives/qt-4.8/qclipboard.html
[qt-svgrendering]: https://doc.qt.io/qt-5/svgrendering.html
[qr-code-gen]: https://github.com/nayuki/QR-Code-generator

**Comment by koutoftimer** [comment]
[Deploying to 3 major platforms][platforms]:
1. X11 Platforms
2. Windows
3. macOS

required, but help with macOS is needed.

[platforms]: https://doc.qt.io/qt-5/deployment.html#platform-specific-notes

**Comment by koutoftimer** [comment]
Debian Stretch [sqrc.zip](https://github.com/koutoftimer/sqrc/files/2955107/sqrc.zip)

**Comment by koutoftimer** [comment]
Linked with static Qt library. Should now run without any external dependencies [sqrc-static.zip](https://github.com/koutoftimer/sqrc/files/2986107/sqrc-static.zip)

**Comment by drhayes** [comment]
Both of those error out with `./sqrc: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.12' not found (required by ./sqrc)`.

**Comment by koutoftimer** [comment]
@drhayes yes. There is one more problem I have figured out few hours ago. It is about Cmake support of Qt's static build.

You may find out more about this in the [qt's developers mailing list](https://lists.qt-project.org/pipermail/development/2015-October/023601.html)

TL;DR: do not use Cmake to build your project with static Qt, use qmake ONLY. 

I have already done everything I just need to commit changes and to release both win and linux versions.

**Comment by koutoftimer** [comment]
@drhayes 
Regular linux executable [sqrc-qt.gz](https://github.com/koutoftimer/sqrc/files/2992942/sqrc-qt.gz)
UPX compressed linux executable [sqrc-qt-upx.gz](https://github.com/koutoftimer/sqrc/files/2992947/sqrc-qt-upx.gz)
Windows executable  [sqrc-qt.zip](https://github.com/koutoftimer/sqrc/files/2992982/sqrc-qt.zip)

**Comment by koutoftimer** [comment]
Sad news. Linux executables depends on following shared libraries. Make sure that your version is [ABI ][abi-wiki] compatible. 
```
libicui18n.so.57 => not found
       libicuuc.so.57 => not found
       libicudata.so.57 => not found
```

Good news is that we have [AppImage][app-image] and I'm starting research in its direction.

[abi-wiki]: https://en.wikipedia.org/wiki/Application_binary_interface
[app-image]: https://appimage.org/

**Comment by koutoftimer** [comment]
Very first linux AppImage available: [SQRC-x86_64.AppImage.tar.gz](https://github.com/koutoftimer/sqrc/files/2999066/SQRC-x86_64.AppImage.tar.gz)

**Comment by koutoftimer** [comment]
Epic fail: I have missed [electron "clone"][clipboard-to-qrcode] somehow. No need to continue this project.

[clipboard-to-qrcode]: https://github.com/ramonreschke/clipboard-to-qrcode
